actionview/lib/action_view/renderer/template_renderer.rb

```diff
@@ -26,7 +26,12 @@ def determine_template(options)
         elsif options.key?(:html)
           Template::HTML.new(options[:html], formats.first)
         elsif options.key?(:file)
-          @lookup_context.with_fallbacks.find_file(options[:file], nil, false, keys, @details)
+          if File.exist?(options[:file])
+            Template::File.new(options[:file])
```

Comment:
Would not this render any file in the filesystem? Would that not add an attack vector on `render file: params[:file]`?

Reply:
That's already an attack vector. `render file:` is the unsafe version of `render template:`

This has me thinking, though. Maybe we could just _fully_ deprecate `render file:`. As far as I can tell it's basically only used to render `public/404.html` and similar pages. Those uses could probably be replaced with either:
* `send_file("#{Rails.root}/public/404.html", status: 404)` or
* `render html: File.read("#{Rails.root}/public/404.html", status: 404)`

Reply:
Yeah. I agree about deprecating it. Or we can change render file to only render things relative to the rails root. It is still an attack vector though since you can render the secrets file if you have it open.